webui/react/src/utils/data.test.ts

```diff
@@ -136,6 +141,58 @@ describe('data utility', () => {
     });
   });
 
+  describe('chained object manipulators', () => {
+    let config = clone(testConfig);
+
+    beforeAll(() => {
+      config = clone(testConfig);
+    });
+
+    describe('getSubObject', () => {
+      it('should return undefined for bad paths', () => {
+        const actual = getSubObject(config, [ 'x', 'y', 'z' ]);
+        expect(actual).toBeUndefined();
+      });
+
+      it('should return null', () => {
+        const actual = getSubObject(config, [ 'min_checkpoint_period' ]);
+        expect(actual).toBeNull();
+      });
+
+      it('should return objects', () => {
+        const actual = getSubObject(config, [ 'searcher' ]);
+        expect(actual).toHaveProperty('mode');
+        expect(typeof actual).toEqual('object');
+      });
+
+      it('should return a reference', () => {
+        const searcher = getSubObject(config, [ 'searcher' ]);
+        const TEST_VALUE = 'TEST';
+        expect(searcher).toHaveProperty('mode');
+        searcher.mode = TEST_VALUE;
+        expect(config.searcher.mode).toEqual(TEST_VALUE);
+      });
+    });
+
+    describe('deleteSubObject', () => {
+      it('should remove from input', () => {
+        expect(config.min_validation_period).not.toBeUndefined();
+        deleteSubObject(config, [ 'min_validation_period' ]);
+        expect(config.min_validation_period).toBeUndefined();
+      });
+    });
+
+    describe('setSubObject', () => {
+      it('should set on input', () => {
+        const value = { abc: 3 };
+        setSubObject(config, [ 'min_validation_period' ], value);
+        expect(config.min_validation_period).toStrictEqual(value);
+        expect(config.min_validation_period === value).toBeTruthy();
```

Comment:
**question:** is this test necessary if the previous line passes?

Reply:
hmm based on the description of the toStrictEqual I wasn't sure if it would reference match the objects. I guess only the second expect would be enough. but the one on 189 could give us more info if the test fails.